proxy/src/transport/tls/config.rs

```diff
@@ -110,27 +168,56 @@ impl CommonSettings {
             })
     }
 
-    /// Stream changes to the files described by this `CommonSettings`.
-    ///
-    /// This will poll the filesystem for changes to the files at the paths
-    /// described by this `CommonSettings` every `interval`, and attempt to
-    /// load a new `CommonConfig` from the files again after each change.
-    ///
-    /// The returned stream consists of each subsequent successfully loaded
-    /// `CommonSettings` after each change. If the settings could not be
-    /// reloaded (i.e., they were malformed), nothing is sent.
-    ///
-    /// TODO: On Linux, this should be replaced with an `inotify` watch when
-    ///       available.
-    pub fn stream_changes(self, interval: Duration)
-        -> impl Stream<Item = CommonConfig, Error = ()>
+    #[cfg(target_os = "linux")]
+    fn stream_changes_inotify(&self)
+        -> Result<impl Stream<Item = (), Error = ()>, Error>
     {
-        self.change_timestamps(interval)
-            .filter_map(move |_|
-                CommonConfig::load_from_disk(&self)
-                    .map_err(|e| warn!("error reloading TLS config: {:?}, falling back", e))
-                    .ok()
-            )
+        use inotify::{Inotify, WatchMask};
+        // NOTE: If we used a less broad watch mask, we could probably avoid
+        //       reloading the certs multiple times when k8s modifies a
+        //       ConfigMap (a multi-step process that we see as a series of
```

Comment:
References to "ConfigMap" should be "ConfigMap/Secret". The private key and end-entity certificate will be stored in a secret.